scripts/tools/initialize_virtualenv.py

```diff
@@ -0,0 +1,128 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import os
+import shutil
+import subprocess
+import sys
+from pathlib import Path
+
+
+def clean_up_airflow_sources(airflow_sources: Path):
+    if airflow_sources.exists():
+        print(f"Removing {airflow_sources}")
+        shutil.rmtree(airflow_sources, ignore_errors=True)
+
+
+def check_if_in_virtualenv() -> bool:
+    return hasattr(sys, 'real_prefix') or (hasattr(sys, 'base_prefix') and sys.base_prefix != sys.prefix)
+
+
+def pip_install_requirements() -> int:
+    """
+    install the requirements of the current python version.
+    return 0 if success, anything else is an error.
+    """
+    version = get_python_version()
+    constraint = (
+        f"https://raw.githubusercontent.com/apache/airflow/constraints-main/constraints-{version}.txt"
+    )
+    pip_install_command = ["pip", "install", "-e", ".[devel-all]", "--constraint", constraint]
+    e = subprocess.run(pip_install_command)
+    return e.returncode
+
+
+def get_python_version() -> str:
+    """
+    return the version of python we are running.
+    """
+    major = sys.version_info[0]
+    minor = sys.version_info[1]
+    return f"{major}.{minor}"
+
+
+def main():
+    """
+    Setup local virtual environment.
+    """
+    airflow_home_dir = os.environ.get("AIRFLOW_HOME", Path.home() / "airflow")  # TODO: change to airflow
+    airflow_sources = str(Path.cwd())
+    if not check_if_in_virtualenv():
+        print(
+            "Local virtual environment not activated.\nPlease create and activate it "
+            "first. (for example using 'python3 -m venv venv && source venv/bin/activate')"
+        )
+        return
```

Comment:
```suggestion
        sys.exit(1)
```

If someone automates the script this will indicate that there was an error. Additionally error code might indicate the reason for error.